mention pro when view not found for fields as a suggestion

pxpm opened the pull request:
## WHY

### BEFORE - What was wrong? What was happening before this PR?

We only said to dev that the view was not found. That was confusing for devs updating form 4 > 5 because the views "existed" in the previous version and now it was not clear why they didn't exist anymore.

### AFTER - What is happening after this PR?

Changed message a bit so it mentions the PRO fields as a suggestion on why view was not found.

tabacitu commented on src/app/Library/CrudPanel/Traits/Views.php:
```diff
@@ -308,6 +308,6 @@ public static function getFirstFieldView($viewPath, $viewNamespace = false)
 
         // if no view exists, in any of the directories above... no bueno
-        abort(500, "Cannot find '{$viewPath}' field view in any of the regular locations.");
+        abort(500, 'Cannot find '.$viewPath.' field view. Was it a typo? Trying to use a <a href="https://backpackforlaravel.com/products/pro-for-unlimited-projects" target="_blank">PRO field?</a>');
```
```suggestion
        abort(500, 'Cannot find the field view: '.$viewPath.'. Please check for typos. Alternatively, maybe you're trying to use a <a href="https://backpackforlaravel.com/products/pro-for-unlimited-projects" target="_blank">PRO field</a>?');
```

tabacitu commented:
Excellent! Changed the wording a little bit and then realised... we may be able to do even better... We have the `backpack_pro()` helper, which tells us if PRO is installed. So we could echo different messages depending on if PRO is installed:
- if PRO is NOT installed - message saying maybe they're trying to use a PRO field (current one);
- if PRO is installed - the old message;

What do you think @pxpm ?

tabacitu commented on src/app/Library/CrudPanel/Traits/Views.php:
```diff
@@ -308,6 +308,6 @@ public static function getFirstFieldView($viewPath, $viewNamespace = false)
 
         // if no view exists, in any of the directories above... no bueno
-        abort(500, "Cannot find '{$viewPath}' field view in any of the regular locations.");
+        abort(500, 'Cannot find the field view: '.$viewPath.'. Please check for typos. Alternatively, maybe you're trying to use a <a href="https://backpackforlaravel.com/products/pro-for-unlimited-projects" target="_blank">PRO field</a>?');
```
```suggestion
        abort(500, 'Cannot find the field view: '.$viewPath.'. Please check for typos. Alternatively, maybe you are trying to use a <a href="https://backpackforlaravel.com/products/pro-for-unlimited-projects" target="_blank">PRO field</a>?');
```

pxpm commented:
Hey @tabacitu !

I think I've addressed your concerns in #4196 . This with the operations had been a wild wild ride! 🙃 Hope you like the solution I came up with. 

fields - throw exception if not pro and no view found, otherwise use default error
chart widget - throw exception if not pro and no view found, otherwise use default error
filters - throw exception if not pro
List - export buttons - throw exception if not pro
List - details row - throw exception if not pro
List - PRO columns - **nothing** - defaults to text column when nothing else found as before
PRO operations (eg Clone) - throw exception if not pro

pxpm commented:
@tabacitu just an heads up about this.

I removed the introduced feature "initializeTraits" and used the `setupXxxDefaults` 👍 

I used the native `is_pro` to check instead of the `trait_exists`. It cleaned the code ALOT.

tabacitu commented on src/app/Exceptions/BackpackProRequiredException.php:
```diff
@@ -0,0 +1,25 @@
+        // 1 use the provided message in full
+        switch ($this->getCode()) {
+            case 0:
+                $this->message = $this->message.' is a Backpack PRO feature. Please purchase and install <a href="https://backpackforlaravel.com/pricing">Backpack\PRO</a>.';
```
![tv4-problem-tom-hanks-houston-we-have-a-4Hx5nJBfi8FzFWxztb](https://user-images.githubusercontent.com/1032474/163144309-1a6fa7b3-74f5-43f8-ab8b-0fedbe661e81.gif)

The lastest error views output `e($exception->getMessage())` so the URL won't work...

tabacitu replied:
So let's remove the URL entirely, in favor of something like:
```suggestion
                $this->message = $this->message.' is a Backpack PRO feature. Please purchase and install the Backpack\PRO addon from backpackforlaravel.com';
```

tabacitu commented on src/app/Exceptions/BackpackProRequiredException.php:
```diff
@@ -0,0 +1,25 @@
+                break;
+        }
+
+        return response(view('errors.500', ['exception' => $this]), 500);
```
Alternatively, instead of throwing a general 500, we could load a custom error view, that lives inside the Backpack repo (eg. `backpack::backpack_error`)... Then over there, we could have additional stuff like buttons after the error message... or something. We could also customize the look & feel of this page, to stand out and provide more information, since normally it's the developer who sees it.

Alternatively, we could maybe throw a Flare error, but just with our custom error message...

What do you think @pxpm ?

tabacitu replied:
I think for now we can just go ahead and remove the links. Then later on, when we have more time, we can create a cool error view, that allows you to pass more information like buttons and such. We'll think of what exactly an exception page needs then.

pxpm replied:
I agree with you. I think it would be WAY WAY more usefull in this cases where the problem is an "Application setup problem". We can have extra links, we can even have a "small" api on that page that could fetch the "articles/faq" that contain the keywords of the reported error.

It would also clearly differentiate from a server error (500) and a Backpack setup error. 

Hope we do it! 🙏

tabacitu commented on src/app/Http/Controllers/ChartController.php:
```diff
@@ -2,6 +2,8 @@
 
-class ChartController extends \Backpack\Pro\Http\Controllers\ChartController
-{
+if (backpack_pro()) {
+    class ChartController extends \Backpack\Pro\Http\Controllers\ChartController
+    {
+    }
 }
```
Why no error-throwing here?

tabacitu replied:
Oh ok! Great.

tabacitu commented on src/app/Library/CrudPanel/Traits/Views.php:
```diff
@@ -308,6 +310,9 @@ public static function getFirstFieldView($viewPath, $viewNamespace = false)
 
         // if no view exists, in any of the directories above... no bueno
+        if (! backpack_pro()) {
+            throw new BackpackProRequiredException('Cannot find the field view: '.$viewPath.'. Please check for typos. Alternatively, maybe you are trying to use a <a href="https://backpackforlaravel.com/products/pro-for-unlimited-projects" target="_blank">PRO field</a>?', 1);
```
Same thing here - cannot use links yet:

```suggestion
            throw new BackpackProRequiredException('Cannot find the field view: '.$viewPath.'. Please check for typos.' . (backpack_pro() ? '' : ' If you are trying to use a PRO field, please first purchase and install the backpack/pro addon from backpackforlaravel.com'), 1);
```

tabacitu commented on src/app/Library/Widget.php:
```diff
@@ -143,8 +144,11 @@ public function getFinalViewPath()
-        return config('backpack.base.component_view_namespaces.widgets')[0].'.'.$this->type;
+        // if no view exists, in any of the directories above... no bueno
+        if (! backpack_pro()) {
+            throw new BackpackProRequiredException('Cannot find the view for «'.$this->type.'» widget type. Please check for typos. Alternatively, maybe you are trying to use a <a href="https://backpackforlaravel.com/products/pro-for-unlimited-projects" target="_blank">PRO widget</a>?', 1);
```
```suggestion
            throw new BackpackProRequiredException('Cannot find the widget view: '.$viewPath.'. Please check for typos.' . (backpack_pro() ? '' : ' If you are trying to use a PRO widget, please first purchase and install the backpack/pro addon from backpackforlaravel.com'), 1);
```

tabacitu reviewed (changes requested):
I love it! Great job here @pxpm , especially with the operation errors - very VERY good solution there.

I just found one problem - and have a simple solution about it. If you're ok with it, I'll go ahead and merge this - can't wait!

pxpm commented:
@tabacitu merged your suggestions, and answered you questions. Let me know if there is something more we should do here. 

Cheers
